lib/open_mic.rb

```diff
@@ -0,0 +1,38 @@
+require 'pry'
+class OpenMic
+  attr_reader :location, :date, :performers
+
+  def initialize(info)
+    @location = info[:location]
+    @date = info[:date]
+    @performers = []
+    @performers_jokes = []
+    @performers_jokes_id = []
+  end
+
+  def welcome(user)
+    @performers << user
+  end
+
+  def repeated_jokes?
+    @performers.each do |performer|
+      @performers_jokes = @performers_jokes + performer.jokes
+    end
+    @performers_jokes.each do |joke|
+      @performers_jokes_id << joke.id
+    end
```

Comment:
Great job getting this to work! This was a tricky one, and you found a working solution, which is amazing. 

Couple points:
1) you don't need to make performers_jokes and performers_jokes_id into instance variables. you're only using these variables inside this method, so you should declare these variables locally, inside the method. 

2) There are some Ruby tools that are a little bit more efficient for this--  like this:

```
  def repeated_jokes?
    all_jokes = @performers.map do |user|
      user.jokes
    end

    all_jokes.flatten.length != all_jokes.flatten.uniq.length
  end
```

The logic is really similar to what you did (collect all jokes, then compare the unique length to the original length-- but just using some tools a little bit better suited to the work.

Reply:
Yes.  I realized after the independent review today and seeing Tim do this challenge that I didn't need to use those instance variables because he didn't.  I asked about it, and Tim helped me understand more clearly about when to use them  and when not to. I think I understand now.  However,  I still don't understand flatten at all.  I also haven't practiced with map do yet.  I'll keep working on it.  Thank-you for the feedback!